File: src/main/java/me/delivery/domain/user/model/entity/BaseEntity.java
package me.delivery.domain.user.model.entity;

import lombok.Getter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

@MappedSuperclass
@EntityListeners(value = AuditingEntityListener.class)
@Getter
public class BaseEntity implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    protected long id;

    @CreatedDate
    @Column(columnDefinition = "DATETIME", nullable = false)
    protected LocalDateTime createdAt;

}

File: src/main/kotlin/me/delivery/domain/entity/BaseEntity.kt
package me.delivery.domain.entity

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.io.Serializable
import java.time.LocalDateTime
import javax.persistence.*

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
class BaseEntity (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "BIGINT UNSIGNED", nullable = false)
    open var id: Long = 0,

    @CreatedDate
    @Column(columnDefinition = "DATETIME", nullable = false)
    open var createdAt: LocalDateTime = LocalDateTime.now()
): Serializable